### PR TITLE
Restructure `inst/components`

### LINKED ISSUE #628: Uncompressed sources of JS missing
## Debian packaging requires uncompressed  JS

Debian packaging requires sources for all non-human readable files.  Compressed JS is considered binary without source and thus we need to provide the sources (and a way to build the compressed version from source) to ship software in an official Debian package.  Version 0.5.0 has three new compressed JS files

    inst/components/accordion.min.js
    inst/components/card.min.js
    inst/components/sidebar.min.js

It would be great if you could either include the uncompressed source into your source tarball or at least link to the origin of these files inside the docs (including the version that is used).

Thanks a lot
   Andreas (from Debian R packaging team)

### LINKED ISSUE #642: Put htmldependencies into individual folders
When bundling, we should either remove the sourcemap hint or put the grouped files into individual folders. 

When making an htmldependency, we can not include extra files directly, only through "allFiles = TRUE". So by having them in individual folders, we can expose the source map without copying all other js/map files for each dependency

- [ ] Nest folders
- [ ] Update dependency folder values

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes #628
Fixes #642

Establishes two folders under `inst/components/`: `scss/` and `dist/`. The `scss/` folder contains source Sass files that are included in the `bs_theme()` dependency. 

The `dist/` folder contains the built JavaScript files for each component under new folders `inst/components/dist/{component}/`, where each component includes minified and unminified versions of the JavaScript. We keep source maps for both since the unminified version is less useful without the source map – we might want to revisit this and only include the source map for the unminified versions.

### CODE REVIEW

File: R/bs-theme.R [hunk 1]
```diff
@@ -377,11 +377,14 @@ bootstrap_bundle <- function(version) {
...
     # nav_spacer() CSS (can be removed)
     nav_spacer = sass_layer(
-      rules = sass_file(system_file("nav-spacer/nav-spacer.scss", package = "bslib"))
+      rules = sass_file(path_bslib_component("scss", "nav-spacer.scss"))
     )
   )
 }
 
+path_bslib_component <- function(...) {
+  system.file("components", ..., package = "bslib")
```
Comment (cpsievert) [inline R/bs-theme.R:386]:
```suggestion
  system_file("components", ..., package = "bslib")
```

File: R/utils.R [hunk 2]
```diff
@@ -140,3 +140,9 @@ get_color_contrast <- function(bg_color) {
   )
 }
 
+js_file_names_with_map <- function(name, minified = NULL) {
+  minified <- get_shiny_devmode_option_minified(minified)
+
+  files <- paste0(name, if (minified) ".min" else "", ".js")
+  paste0(files, c("", ".map"))
```
[the remark this answers is not in the document]
Comment (cpsievert) [reply R/utils.R:147]: I think this helper could go away (see https://github.com/rstudio/bslib/pull/658/files#r1254548274)

File: R/bs-theme.R (hunk 1 above)
Comment (cpsievert) [reply R/bs-theme.R:386]: I'm also a little tempted to say that we should abstract and use the current `lib_path()` for this (to get file checking) and call it `inst_path()`, then `lib_path()` and `component_path()` could be implemented as:

```r
lib_path <- function(...) {
  inst_path("lib", ...)
}
components_path <- function(...) {
  inst_path("components", ...)
}
```

File: R/sidebar.R [hunk 3]
```diff
@@ -278,13 +278,16 @@ collapse_icon <- function() {
...
     name = "bslib-sidebar",
     version = get_package_version("bslib"),
     package = "bslib",
-    src = "components",
-    script = "sidebar.min.js"
+    src = "components/dist/sidebar",
+    all_files = FALSE,
+    script = js_file_names_with_map("sidebar", minified = minified)
   )
 }
```
[the remark this answers is not in the document]
Comment (gadenbuie) [reply R/sidebar.R]: Done in cf0ed3a9227f0b7aaedeb48d78bfee1dad0e61a6

File: R/utils.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (gadenbuie) [reply R/utils.R]: Done in cf0ed3a9227f0b7aaedeb48d78bfee1dad0e61a6

File: R/bs-theme.R (hunk 1 above)
Comment (gadenbuie) [reply R/bs-theme.R]: I did this by abstracting the existing `lib_file()` into `inst_file()` and then adding `lib_file()` and `components_file()`.

I like the approach but not the names. I'd prefer `path_inst()`, `path_lib()`, `path_components()`, etc. which would help with internal discoverability.

**Review by cpsievert — APPROVED** [review]

**Pull request merged**